splice_machine/src/test/java/com/splicemachine/foreignkeys/ForeignKeyMetadataIT.java

```diff
@@ -153,6 +157,44 @@ public void testSimplifiedForeignKeysDoesNotExplode() throws Exception{
         }
     }
 
+    @Test
+    public void testReferencesAndSysKeyColUseViews() throws Exception {
+        methodWatcher.executeUpdate("create table if not exists SELF_REF (c int, d int, " +
+                "constraint SRPK primary key (c), constraint SRFK foreign key (c) references SELF_REF(c))");
+
+        String query = "select * from syscat.references r where r.tabname = 'SELF_REF'";
+
+        String expected =
+                "CONSTNAME |      TABSCHEMA      | TABNAME | OWNER | OWNERTYPE |REFKEYNAME |    REFTABSCHEMA     |REFTABNAME |COLCOUNT |DELETERULE |UPDATERULE | CREATE_TIME |\n" +
+                "--------------------------------------------------------------------------------------------------------------------------------------------------------------\n" +
+                "   SRFK    |FOREIGNKEYMETADATAIT |SELF_REF | NULL  |   NULL    |   SRPK    |FOREIGNKEYMETADATAIT | SELF_REF  |    1    |     A     |     A     |    NULL     |";
```

Comment:
could you please add more tests for the following cases:

- foreign key with delete rules and update rules.
- table which is owned by a user (affecting `OWNER` table I guess)

Reply:
Unused columns are removed and added test cases on different on delete/update rules.